### PR TITLE
fix: allow extend setter or getter alone

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/loader/mixin/extend.js [hunk 1]
```diff
@@ -96,7 +103,25 @@ module.exports = {
         }
 
         // Copy descriptor
-        const descriptor = Object.getOwnPropertyDescriptor(ext, property);
+        let descriptor = Object.getOwnPropertyDescriptor(ext, property);
+        let orginalDescriptor = Object.getOwnPropertyDescriptor(proto, property);
```
Comment (fengmk2) [inline lib/loader/mixin/extend.js]: Object.create(obj) 导致 getOwnPropertyDescriptor 失败了

**Comment by popomore** [comment]
+ egg-core@3.2.1

File: lib/loader/mixin/extend.js [hunk 2]
```diff
@@ -96,7 +103,25 @@ module.exports = {
...
+        if (!orginalDescriptor) {
+          // try to get descriptor from orginalPrototypes
+          const orginalProto = orginalPrototypes[name];
+          if (orginalProto) {
+            orginalDescriptor = Object.getOwnPropertyDescriptor(orginalProto, property);
```
Comment (dead-horse) [inline lib/loader/mixin/extend.js]: 这里只判断了 koa 的，egg 框架和插件互相覆盖还是不行？

Comment (popomore) [inline lib/loader/mixin/extend.js]: 下面 define 一次后，就可以取到了。

**Pull request merged**